elements/lisk-p2p/src/p2p.ts

```diff
@@ -365,7 +364,36 @@ export class P2P extends EventEmitter {
 			this.emit(EVENT_INBOUND_SOCKET_ERROR, error);
 		};
 
-		this._peerPool = new PeerPool({
+		const peerPoolConfig = P2P._createPeerPoolConfig(config);
+		this._peerPool = new PeerPool(peerPoolConfig);
+
+		this._bindHandlersToPeerPool(this._peerPool);
+		// Add peers to tried peers if want to re-use previously tried peers
+		if (this._sanitizedPeerLists.previousPeers) {
+			this._sanitizedPeerLists.previousPeers.forEach(peerInfo => {
+				if (!this._peerBook.getPeer(peerInfo)) {
+					this._peerBook.addPeer(peerInfo);
+					this._peerBook.upgradePeer(peerInfo);
+				} else {
+					this._peerBook.upgradePeer(peerInfo);
+				}
+			});
+		}
+
+		this._nodeInfo = config.nodeInfo;
+		this.applyNodeInfo(this._nodeInfo);
+
+		this._populatorInterval = config.populatorInterval
+			? config.populatorInterval
+			: DEFAULT_POPULATOR_INTERVAL;
+
+		this._peerHandshakeCheck = config.peerHandshakeCheck
+			? config.peerHandshakeCheck
+			: validatePeerCompatibility;
+	}
+
+	private static _createPeerPoolConfig(config: P2PConfig): PeerPoolConfig {
```

Comment:
why we are making it a `static` function if it's not a `public` function, it doesn't add any value to make it `static`

Reply:
If it does not use the context, please use pure function

Reply:
Adressed.